tests/test_assert.h

```diff
@@ -54,14 +54,55 @@ void TestFail(const char *expval, const char *val, const char *exp,
               const char *file, int line, const char *func = 0);
 
 void TestEqStr(const char *expval, const char *val, const char *exp,
-               const char *file, int line);
+               const char *file, int line, const char *func = 0);
+
+// Workaround for `enum class` printing.
+// There is an issue with the printing of enums with a fixed underlying type.
+// These enums are generated by `flatc` if `--scoped-enums` is active.
+// All modern compilers have problems with `std::stringstream&<<(T v)` if T is
+// an enum with fixed type. For details see DR1601:
+// http://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1601
+// https://stackoverflow.com/questions/34336024/ambiguous-overload-when-writing-an-enum-with-an-enum-base-but-only-with-clang
+
+template<typename T, bool is_enum_type = flatbuffers::is_enum<T>::value>
+struct underlying_of_scalar {
+  static_assert(flatbuffers::is_scalar<T>::value, "invalid type T");
+  typedef T type;
+};
+
+template<typename T> struct underlying_of_scalar<T, true> {
+// clang-format off
+  // There are old compilers without full C++11 support (see stl_emulation.h).
+  #if defined(FLATBUFFERS_TEMPLATES_ALIASES) && !defined(FLATBUFFERS_CPP98_STL)
+  using type = typename std::underlying_type<T>::type;
+  #else
+  typedef int64_t type;
```

Comment:
I think `int32_t` for enums was more common on older compilers, but it probably doesn't matter.

Reply:
MSVC2010 doesn't support non-fixed enums bigger than `int`, but gcc and clang do.
But you are right, it does not matter for the tests.